From: =?utf-8?q?IOhannes_m_zm=C3=B6lnig?= <umlaeute@debian.org>
Date: Sun, 11 Feb 2018 00:00:00 +0100
Subject: clamp a few defines to fixed values

 on Debian we only ever allow using the system provided libraries
 (zlib, flac, oggvorbis,...)
Origin: Debian
Forwarded: not-needed
---
 .../utility/juce_CheckSettingMacros.h              | 25 +++++++++++++++++
 modules/juce_core/juce_core.h                      | 32 +++++++++++++++++++++-
 2 files changed, 56 insertions(+), 1 deletion(-)

diff --git a/modules/juce_audio_plugin_client/utility/juce_CheckSettingMacros.h b/modules/juce_audio_plugin_client/utility/juce_CheckSettingMacros.h
index b5278b1..4c0a9ba 100644
--- a/modules/juce_audio_plugin_client/utility/juce_CheckSettingMacros.h
+++ b/modules/juce_audio_plugin_client/utility/juce_CheckSettingMacros.h
@@ -24,6 +24,30 @@
   ==============================================================================
 */
 
+/* Debian specific overrides */
+#if JucePlugin_Build_LV2 && ! defined (JucePlugin_LV2URI)
+# warning JUCE/Debian using default LV2URI
+# define JucePlugin_LV2URI               JucePlugin_Name
+#endif
+
+#if JucePlugin_Build_AU
+# warning JUCE/Debian disabled AU support
+# undef JucePlugin_Build_AU
+# define JucePlugin_Build_AU 0
+#endif
+
+#if JucePlugin_Build_RTAS
+# warning JUCE/Debian disabled RTAS support
+# undef JucePlugin_Build_RTAS
+# define JucePlugin_Build_RTAS 0
+#endif
+
+#if JucePlugin_Build_AAX
+# warning JUCE/Debian disabled AAX support
+# undef JucePlugin_Build_AAX
+# define JucePlugin_Build_AAX 0
+#endif
+
 // The following checks should cause a compile error if you've forgotten to
 // define all your plugin settings properly..
 
@@ -40,6 +64,7 @@
 #endif
 #define JUCE_CHECKSETTINGMACROS_H
 
+
 #ifndef JucePlugin_IsSynth
  #error "You need to define the JucePlugin_IsSynth value!"
 #endif
diff --git a/modules/juce_core/juce_core.h b/modules/juce_core/juce_core.h
index 9cd4146..e26e139 100644
--- a/modules/juce_core/juce_core.h
+++ b/modules/juce_core/juce_core.h
@@ -64,6 +64,36 @@
 
 #include "system/juce_TargetPlatform.h"
 
+/** Debian specific constants
+  - Debian, we don't use embedded libraries
+*/
+#ifdef JUCE_INCLUDE_ZLIB_CODE
+# undef JUCE_INCLUDE_ZLIB_CODE
+#endif
+#define JUCE_INCLUDE_ZLIB_CODE 0
+
+#ifdef JUCE_INCLUDE_FLAC_CODE
+# undef JUCE_INCLUDE_FLAC_CODE
+#endif
+#define JUCE_INCLUDE_FLAC_CODE 0
+
+#ifdef JUCE_INCLUDE_OGGVORBIS_CODE
+# undef JUCE_INCLUDE_OGGVORBIS_CODE
+#endif
+#define JUCE_INCLUDE_OGGVORBIS_CODE 0
+
+#ifdef JUCE_INCLUDE_JPEGLIB_CODE
+# undef JUCE_INCLUDE_JPEGLIB_CODE
+#endif
+#define JUCE_INCLUDE_JPEGLIB_CODE 0
+
+#ifdef JUCE_INCLUDE_PNGLIB_CODE
+# undef JUCE_INCLUDE_PNGLIB_CODE
+#endif
+#define JUCE_INCLUDE_PNGLIB_CODE 0
+/** Debian specifics END
+*/
+
 //==============================================================================
 /** Config: JUCE_FORCE_DEBUG
 
@@ -123,7 +153,7 @@
     specify the path where your zlib headers live.
 */
 #ifndef JUCE_INCLUDE_ZLIB_CODE
- #define JUCE_INCLUDE_ZLIB_CODE 1
+ #define JUCE_INCLUDE_ZLIB_CODE 0
 #endif
 
 #ifndef JUCE_ZLIB_INCLUDE_PATH
